pkg/network/usm/istio.go

```diff
@@ -285,11 +290,17 @@ func (m *istioMonitor) getEnvoyPath(pid uint32) string {
 	}
 
 	buffer = buffer[:n]
-	i := bytes.Index(buffer, envoyCmd)
+	envoyPath, err := os.Readlink(m.envoyCmd)
```

Comment:
that's wrong
you should build the path `/proc/<pid>/exe` and run readlink on that

Reply:
changed